lib/ansible/modules/packaging/os/swupd.py

```diff
@@ -0,0 +1,286 @@
+description:
+  - "Manages bundles with the M(swupd) update manager, which is used by the
+    Clear Linux Project for Intel Architecture."
+version_added: "2.3"
+author: Alberto Murillo (@albertomurillo)
+options:
+  name:
+    description:
+      - Name of the (C)bundle to install or remove.
+    required: false
+    default: null
+    aliases: ['bundle']
+  state:
+    description:
+      - Indicates the desired (C)bundle state. I(present) ensures the bundle
+        is installed while I(absent) ensures the (C)bundle is not installed.
+    required: false
+    default: present
+    choices: ["present", "absent"]
+  update:
+    description:
+      - Updates the OS to the latest version.
+    required: false
+    default: no
+  verify:
+    description:
+      - Verify content for OS version.
+    required: false
+    default: null
+  manifest:
+    description:
+      - Manifest to verify against to.
+    required: false
+    default: null
+    aliases: ['release', 'version']
+  format:
+    description:
+      - The format suffix for version file downloads.
+    required: false
+    default: null
+  url:
+    description:
+      - Overrides both C(contenturl) and C(versionurl).
+    required: false
+    default: null
+  contenturl:
+    description:
+      - URL for content file download.
+    required: false
+    default: null
+  versionurl:
+    description:
+      - URL for version string download.
+    required: false
+    default: null
+'''
+
+EXAMPLES = '''
+- name Update the OS to the latest version
+  swupd:
+    update: yes
+
+- name: Installs the "foo" bundle
+  swupd:
+    name: foo
+    state: present
+
+- name: Removes the "foo" bundle
+  swupd:
+    name: foo
+    state: absent
+
+- name: Check integrity of filesystem
+  swupd:
+    verify: yes
+
+- name: Downgrade OS to release 12920
+  swupd:
+    verify: yes
+    manifest: 12920
+'''
+
+FILES_NOT_MATCH = "files did not match"
+FILES_REPLACED = "missing files were replaced"
+FILES_FIXED = "files were fixed"
+FILES_DELETED = "files were deleted"
+
+
+def _get_cmd(module, command):
+    p = module.params
+    cmd = "%s %s" % (SWUPD_CMD, command)
+
+    if p["format"]:
+        cmd += " --format=%s" % p["format"]
+    if p["manifest"]:
+        cmd += " --manifest=%s" % p["manifest"]
+    if p["url"]:
+        cmd += " --url=%s" % p["url"]
+    else:
+        if p["contenturl"] and command != "check-update":
+            cmd += " --contenturl=%s" % p["contenturl"]
+        if p["versionurl"]:
+            cmd += " --versionurl=%s" % p["versionurl"]
+
+    return cmd
+
+
+def _is_bundle_installed(module, bundle):
+    try:
+        os.stat("/usr/share/clear/bundles/%s" % bundle)
+    except OSError:
+        return False
+
+    return True
+
+
+def _needs_update(module):
+    cmd = _get_cmd(module, "check-update")
+    rc, stdout, stderr = module.run_command(cmd, check_rc=False)
+
+    if rc == 0:
+        return True
+
+    if rc == 1:
```

Comment:
I think this style is more readable, there is no need for another variable for such a simple function